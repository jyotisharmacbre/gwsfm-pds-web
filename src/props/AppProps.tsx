export interface IAppProps{
    Theme: any,
    UseStyles: any
}

export interface IHeaderPage
{
    Title:string,
    ActionList?: IBtnActionProps []
}

export interface IBtnActionProps
{
    Title: string,
    Icon: any,
    Css?: any,
    HandleClick:  ()=> void
}