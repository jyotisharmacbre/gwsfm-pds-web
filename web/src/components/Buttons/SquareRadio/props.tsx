export interface ISquareRadioProps{
    value:string;
    label:string;
    id:string;
}