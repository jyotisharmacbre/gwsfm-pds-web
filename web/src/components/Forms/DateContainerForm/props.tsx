export interface IDateContainerProps{
dateOneLabel: string;
dateTwoLabel: string;
}