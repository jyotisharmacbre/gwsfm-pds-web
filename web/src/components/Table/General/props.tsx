export interface IGeneralTableProps {
  headers: IGeneralTableHeaderProps[];
  content: string;
  editActionClick: () => void;
}

export interface IGeneralTableHeaderProps {
  heading: string;
  subHeading: string;
}
