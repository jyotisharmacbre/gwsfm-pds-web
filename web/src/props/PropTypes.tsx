export namespace AppTypes
{
    export type Color = 'cbregreen' | 'primary' | 'secondary' | 'save' | 'back'  | undefined;
}