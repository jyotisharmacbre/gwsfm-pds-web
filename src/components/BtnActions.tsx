import React from 'react';
import { Button, Box, makeStyles, Theme, createStyles, IconButton } from '@material-ui/core';
import { IBtnActionProps } from '../props/AppProps';
import { Add, LibraryBooks } from '@material-ui/icons';

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        buttonPrimary: {
            margin: theme.spacing(1),
            color: 'white',
            backgroundColor:'black'
        },
        buttonSecondary: {
            margin: theme.spacing(1),
            borderColor: 'green',
            color: '#00684d'
        },
        leftIcon: {
            marginRight: theme.spacing(1),
            
        },
        rightIcon: {
            marginLeft: theme.spacing(1),
        },
        iconSmall: {
            fontSize: 20,
        },
    }),
);

function GetIcons(icon: string) {
    const styles = useStyles();
    switch (icon) {
        case "create":
            return (<Add className={styles.leftIcon} />);
        case "pipeline":
            return (<LibraryBooks className={styles.leftIcon} />);
    }
}

export default function PageActions(props: { Actions: IBtnActionProps[] }) {
    const styles = useStyles();
    const items = props.Actions.map((item) => 
        <Button key={item.Title} variant="contained" color="primary" className={item.Icon == 'create'? styles.buttonPrimary : styles.buttonSecondary} onClick={item.HandleClick} >
            {GetIcons(item.Icon)}
            {item.Title}
        </Button>
    );

    return (
        <React.Fragment >
            {items}
        </React.Fragment>);
}



