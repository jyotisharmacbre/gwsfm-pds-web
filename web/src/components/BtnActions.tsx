import React from 'react';
import { Button, makeStyles, Theme, createStyles } from '@material-ui/core';
import { IBtnActionProps } from '../props/AppProps';
import { AppTypes } from '../props/PropTypes';
import { Backspace, AddCircleOutline, LibraryBooks, ViewList } from '@material-ui/icons';

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        buttonPrimary: {
            margin: theme.spacing(1),
            color: 'white',
            backgroundColor: 'black'
        },
        buttonSecondary: {
            margin: theme.spacing(1),
            color: '#ffffff',
            backgroundColor: '#00684d'
        },
        buttonSave: {
            margin: theme.spacing(1),
            backgroundColor: '#D3D3D3',
            color: '#000000'
        },
        buttondefault: {
            margin: theme.spacing(1),
            backgroundColor: 'white',
            color: '#000000'
        },
        buttonGreen: {
            margin: theme.spacing(1),
            color: 'white',
            backgroundColor: '#00684d'
        },
        leftIcon: {
            marginRight: theme.spacing(1),
            fontSize: 20,
        },
        rightIcon: {
            marginLeft: theme.spacing(1),
        },
        iconSmall: {
            fontSize: 20,
        },
    }),
);

const GetIcons = (icon: string) => {
    const styles = useStyles();
    switch (icon) {
        case "create":
            return (<AddCircleOutline className={styles.leftIcon} />);
        case "pipeline":
            return (<LibraryBooks className={styles.leftIcon} />);
        case "backspace":
            return (<Backspace className={styles.leftIcon} />);
        case "viewall":
            return (<ViewList className={styles.leftIcon} />)
        default:
            return;
    }
}

const GetButtonStyle = (type: AppTypes.Color) => {
    const styles = useStyles();
    switch (type) {
        case 'primary':
            return styles.buttonPrimary;
        case 'secondary':
            return styles.buttonSecondary;
        case 'save':
            return styles.buttonSave;
        case 'cbregreen':
            return styles.buttonGreen;
        default:
            return styles.buttondefault;
    }
}

export function PageBtnActions(props: { Actions: IBtnActionProps[] }) {
    const items = props.Actions.map((item) =>
        <Button type="submit" key={item.Title} variant='contained' color={(item.Color ? undefined : 'primary')} className={GetButtonStyle(item.Color || undefined)} onClick={item.HandleClick}  >
            {GetIcons(item.Icon)}
            {item.Title}
        </Button>
    );

    return (
        <React.Fragment >
            {items}
        </React.Fragment>);
}



