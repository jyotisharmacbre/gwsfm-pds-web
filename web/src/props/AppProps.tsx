import { AppTypes } from "./PropTypes";
import { IProjectForm, IProjectFormAddAction } from "../session/ProjectForm/Type";

export interface IAppProps {
    Theme: any,
    UseStyles: any
}

export interface IHeaderPageProps {
    Title: string,
    ActionList?: IBtnActionProps[]
}

export interface IBtnActionProps {
    Title: string,
    Icon?: any,
    Color?: AppTypes.Color,
    HandleClick: () => void
}

export interface IProjectFormState {
    readonly projectname: string,
    readonly company: string;
    readonly customer_contract: string,
    readonly pmexperience: boolean;
    readonly projectmanager: string;
    readonly projectscope: string;
    readonly locale: string;
    invalidProjectName: boolean;
    invalidProjectManager: boolean;
    invalidCompany: boolean;
    invalidCustomerContract: boolean;
    invalidPMExperience: boolean;
    invalidProjectScope: boolean;
    invalidLocale: boolean;
}

export interface IProjectFormProps {
    handleClick: (data: any) => void;
    addToForm: (data: IProjectForm) => Promise<IProjectFormAddAction>;
    form: IProjectForm;
}
