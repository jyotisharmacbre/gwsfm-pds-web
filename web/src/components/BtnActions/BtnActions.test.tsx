import React from 'react';
import ReactDOM from 'react-dom';
import { PageBtnActions } from './BtnActions';
import { IBtnActionProps } from '../../props/AppProps';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

it('PageBtnActions should render', () => {
    const component = renderer.create(
        <PageBtnActions Actions={[]} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('PageBtnActions should render buttons', () => {
    let btn1: IBtnActionProps = {
        Title: 'Test',
        Icon: 'create',
        HandleClick: () => {
        }
    }

    const component = mount(
        <PageBtnActions Actions={[btn1]} />
    );

    expect(component.find('button').text()).toEqual('Test');
    
 

});
