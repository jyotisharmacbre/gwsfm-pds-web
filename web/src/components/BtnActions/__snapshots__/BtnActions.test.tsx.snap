// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageBtnActions should render 1`] = `null`;
